File: BACKEND-APPLICATION/src/main/java/com/application/entities/Room.java
package com.application.entities;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "rooms")
@Setter
@Getter
public class Room {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String name;
    private String description;
    private String imageUrl;
    private Float price;
    private boolean isAvailable;
    @ManyToOne
    @JoinColumn(name = "hotel_id")
    private Hotel hotel;
    @OneToMany(mappedBy = "room")
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private List<Reservation> reservations = new ArrayList<>();

    public Room() { }

}

File: FRONTEND-APPLICATION/src/app/models/room.ts
import { Hotel } from "./hotel";

export class Room {

  public id: number | undefined;
  public name: string;
  public description: string;
  public imageUrl: string;
  public price: number | undefined;
  public available: boolean;
  public hotel: Hotel ;

  constructor() {
    this.id = undefined;
    this.name = '';
    this.description = '';
    this.imageUrl = '';
    this.price = undefined;
    this.available= true;
    this.hotel = new Hotel();
  }

}